### PR TITLE
Goodbye Stamen

### LINKED ISSUE #796: Goodbye Stamen
Stamen has been removed from the Tiles providers listed in xyzservices by 2023.10.1
https://github.com/geopandas/xyzservices/releases/tag/2023.10.1

This breaks the installation test because `climada.engine.impact.Impact.plot_basemap_XXX_exposure` use `ctx.providers.Stamen.Terrain` as default.

Short term solution (de-corrupt the environment on Jenkins): [pin xyzservices version to 2023.10.0](https://github.com/CLIMADA-project/climada_python/commit/c8a05872c9cd7c67655b269799b123b0489f692f)

Long term solution: pick another default.

### PR DESCRIPTION
**Pull request by emanuel-schmid**
Changes proposed in this PR:
- replace the default tiles provider to use OpenStreetMap instead of StamenTerrain

This PR fixes #796

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [x] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/util/api_client.py
[the remark this answers is not in the document]
Comment (peanutfun) [reply climada/util/api_client.py:1]: Alright, fine 🙃

**Review by peanutfun — APPROVED** [review]
Looks good, thanks for fixing this! ☺️

**Comment by rogerilesdr** [comment]
Hi Lukas, I'm running the latest core CLIMADA release v4.01 on Anaconda which still has the Stamen issue. I was just interested to know when an update with the resolution will be released so I can decide whether to look for a temporary workaround? 
Thanks, Roger

**Comment by peanutfun** [comment]
@rogerilesdr The workaround is to specify a different provider in the `Impact` method parameters, e.g.,
```python
import contextily as ctx
impact.plot_basemap_eai_exposure(url=ctx.providers.CartoDB.Positron)
```

**Comment by peanutfun** [comment]
@gaelleclain @rogerilesdr  On second thought, I am not sure if that fix helps because the code in `impact.py` still includes the provider which does not exist anymore. You might need to use the development version of Climada, following the "Advanced Instructions" for installation until our next release.

**Comment by rogerilesdr** [comment]
That's OK, I've done a dirty workaround for now, I've manually updated the files changed, rather than using the development version for now. I'll overwrite these files when the next stable release arrives.

**Comment by gaelleclain** [comment]
@rogerilesdr can you please give a few details on your workaround?
- Did you update any other file than impact.py?
- Did you do anything else before testing? 

Thanks

**Comment by rogerilesdr** [comment]
I updated the .py files listed on the Files tab above, which also shows where they're located... namely...
- test_plot.py
- impact.py
- base.py
- api_client.py

**Pull request merged**